Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/MATLABAnalytics/ByMATLABAnalytic.tsx

```diff
@@ -0,0 +1,194 @@
+    const [sortField, setSortField] = React.useState<keyof OpenXDA.Types.MATLABAnalytic>('LoadOrder');
+    const [ascending, setAscending] = React.useState<boolean>(true);
+    const [errors, setErrors] = React.useState<string[]>([]);
+    const [status, setStatus] = React.useState<Application.Types.Status>('unintiated');
+
+    let history = useHistory();
+
+    const MATLABAnalyticSearchFields = [
+        { label: 'Method Name', key: 'MethodName', type: 'string', isPivotField: false },
+        { label: 'Assembly Name', key: 'AssemblyName', type: 'string', isPivotField: false },
+        { label: 'Load Order', key: 'LoadOrder', type: 'number', isPivotField: false }
+    ];
+    const MATLABAnalyticDefaultSearchField = { label: 'Method Name', key: 'MethodName', type: 'string', isPivotField: false };
+    const [search, setSearch] = React.useState<Array<Search.IFilter<OpenXDA.Types.MATLABAnalytic>>>([]);
+
+    const emptyRecord = { ID: 0, AssemblyName: '', MethodName: '', SettingSQL: '', LoadOrder: 0 };
+    const [record, setRecord] = React.useState<OpenXDA.Types.MATLABAnalytic>(emptyRecord);
+    const [eventTypeRecord, setEventTypeRecord] = React.useState<OpenXDA.Types.MATLABAnalyticEventType[]>([]);
+    const [assetTypeRecord, setAssetTypeRecord] = React.useState<OpenXDA.Types.MATLABAnalyticAssetType[]>([]);
+
+    React.useEffect(() => {
+        if (analyticStatus == 'unintiated' || analyticStatus == 'changed')
+            dispatch(MATLABAnalyticSlice.DBSearch({ filter: search, sortField, ascending }));
+    }, [analyticStatus]);
+
+    React.useEffect(() => {
+        dispatch(MATLABAnalyticSlice.DBSearch({ filter: search, sortField, ascending }));
+    }, [search, sortField, ascending]);
+
+    React.useEffect(() => {
+        let e = [];
+        if (record.MethodName === null || record.MethodName.length === 0) {
+            e.push('A Method Name is required.');
+        }
+        if (record.AssemblyName === null || record.AssemblyName.length === 0) {
+            e.push('An Assembly Name is required.');
+        }
+
+        setErrors(e);
+    }, [record]);
+
+    function handleSelect(item) {
+        if (props.Roles.indexOf('Administrator') != -1) {
+            history.push({ pathname: homePath + 'index.cshtml', search: '?name=MATLABAnalytic&AnalyticID=' + item.row.ID });
+        } 
+    }
+
+    function addNewAnalytic() {
+        let handle = $.ajax({
+            type: "POST",
+            url: `${homePath}api/OpenXDA/MATLABAnalytic/Add`,
+            contentType: "application/json; charset=utf-8",
+            data: JSON.stringify({
+                MATLABAnalytic: record,
+                MATLABAnalyticEventType: eventTypeRecord,
+                MATLABAnalyticAssetType: assetTypeRecord,
+            }),
+            dataType: 'json',
+            cache: false,
+            async: true
+        }).done(() => {
+            dispatch(MATLABAnalyticSlice.DBSearch({ filter: search, sortField, ascending }));
```

Comment:
```suggestion
            dispatch(MATLABAnalyticSlice.SetChanged());
```
I would recommend using this dispatch. It changes your status on both search and fetch to changed, so that if you ever need fetch you don't need to worry about dispatching here as well, since the hook will catch it.